engine/tests/provider/sql_transpiler_test.cpp

```diff
@@ -0,0 +1,129 @@
+	void SetUp() override {
+    auto get_all_lineitem_col_idxs = [] (size_t column_count) -> std::vector<int> {
+        std::vector<int> ret(column_count);
+        std::iota(ret.begin(), ret.end(), 0);
+        return ret;
+    };
+
+    auto get_ops_without_pars = [] () -> std::map<operator_type, ral::io::sql_tools::operator_info> {
+        auto ret = ral::io::sql_tools::get_default_operators();
+        for (auto& [op, op_info] : ret) {
+            op_info.parentheses_wrap = false;
+        }
+        ral::io::sql_tools::operator_info oi;
+        oi.parentheses_wrap = false;
+        ret[operator_type::BLZ_LOGICAL_AND] = oi;
+        ret[operator_type::BLZ_LOGICAL_OR] = oi;
+        ret[operator_type::BLZ_ADD] = oi;
+        ret[operator_type::BLZ_SUB] = oi;
+        ret[operator_type::BLZ_MUL] = oi;
+        ret[operator_type::BLZ_DIV] = oi;
+        ret[operator_type::BLZ_LESS] = oi;
+        ret[operator_type::BLZ_LESS_EQUAL] = oi;
+        ret[operator_type::BLZ_GREATER] = oi;
+        ret[operator_type::BLZ_GREATER_EQUAL] = oi;
+        return ret;
+    };
+
+    std::vector<std::string> lineitem_cols = {
+      "l_orderkey", "l_partkey", "l_suppkey", "l_linenumber", "l_quantity",
+      "l_extendedprice", "l_discount", "l_tax", "l_returnflag", "l_linestatus",
+      "l_shipdate", "l_commitdate", "l_receiptdate", "l_shipinstruct",
+      "l_shipmode", "l_comment"};
+    auto all_lineitem_col_idxs = get_all_lineitem_col_idxs(lineitem_cols.size());
+    auto ops_without_pars = get_ops_without_pars();
+
+    auto p = GetParam();
+    this->transformer_type = p.transformer_type;
+    this->queryPart = p.queryPart;
+    this->column_names = p.column_names.empty()? std::move(lineitem_cols) : p.column_names;
+    this->column_indices = p.column_indices.empty()? std::move(all_lineitem_col_idxs) : p.column_indices;
+    this->operators = p.operators.empty()? std::move(ops_without_pars) : p.operators;
+    this->expected = p.expected;
+  }
+
+	void TearDown() override {}
+
+protected:
+  DataType transformer_type;
+  std::string queryPart;
+  std::vector<std::string> column_names;
+  std::vector<int> column_indices;
+  std::map<operator_type, sql_tools::operator_info> operators;
+  std::string expected;
+};
+
+const std::vector<SQLTranspilerTestParam> default_check_entries = {
+  SQLTranspilerTestParam(
+    "BindableTableScan(table=[[main, lineitem]], filters=[[<(+($5, $6), 100)]])",
+    "l_extendedprice + l_discount < 100"
+  ),
+  SQLTranspilerTestParam(
+    "BindableTableScan(table=[[main, lineitem]], filters=[[<(+($5, $6), 100)]])",
+    "((l_extendedprice + l_discount) < 100)",
+    sql_tools::get_default_operators()
+  ),
+  SQLTranspilerTestParam(
+    "BindableTableScan(table=[[main, lineitem_MYSQL]], filters=[[AND(>=($10, 1995-01-01), <=($11, 1996-12-31))]], projects=[[0, 2, 5, 6, 10]], aliases=[[l_orderkey, l_suppkey, l_extendedprice, l_discount, l_shipdate]])",
```

Comment:
I think this is incorrect. `projects=[[0, 2, 5, 6, 10]]`  happens first, which means that the columns referred to in filters can only be $0 - $5, but you have $10 and $11.